glusterd2/commands/snapshot/snapshot-status.go

```
@@ -0,0 +1,178 @@
+	ctx.SetNodeResult(gdctx.MyUUID, brickStatusTxnKey, snapshotStatusesResp)
+	return nil
+
+}
+
+func registerSnapshottatusStepFuncs() {
+	transaction.RegisterStepFunc(snapshotStatus, "snap-status.Check")
+}
+
+func snapshotStatusHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	snapname := mux.Vars(r)["snapname"]
+	snap, err := snapshot.GetSnapshot(snapname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, err)
+		return
+	}
+
+	vol := &snap.SnapVolinfo
+	if vol.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrSnapNotActivated)
+		return
+	}
+	txn, err := transaction.NewTxnWithLocks(ctx, vol.Name)
+	if err != nil {
+		if err == transaction.ErrLockTimeout {
+			restutils.SendHTTPError(ctx, w, http.StatusConflict, err)
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		}
+		return
+	}
+
+	defer txn.Done()
+	txn.Steps = []*transaction.Step{
+		{
+			DoFunc: "snap-status.Check",
+			Nodes:  vol.Nodes(),
+		},
+	}
+	txn.Ctx.Set("snapname", snapname)
+
+	// Some nodes may not be up, which is okay.
+	txn.DontCheckAlive = true
+	txn.DisableRollback = true
+
+	err = txn.Do()
+	if err != nil {
+		logger.WithError(err).WithField("snapname", snapname).Error("Failed to get snapshot status")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	resp, err := createSnapshotStatusesResp(txn.Ctx, snap)
+	if err != nil {
+		errMsg := "Failed to aggregate snapshot status results from multiple nodes."
+		logger.WithField("error", err.Error()).Error("BrickStatusHandler:" + errMsg)
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, errMsg)
+		return
+	}
+
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, resp)
+}
+
+func createSnapshotStatusesResp(ctx transaction.TxnCtx, snap *snapshot.Snapinfo) (*api.SnapStatusResp, error) {
```

I don't think you need to return an error here. as you are always returning nil